myfitnesspal/client.py

```diff
@@ -812,3 +818,485 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+    ) -> None:
+        """Function to submit new foods / groceries to the MyFitnessPal database. Function will return True if successful."""
+
+        SUBMIT_PATH = "food/submit"
+        SUBMIT_DUPLICATE_PATH = "food/duplicate"
+        SUBMIT_NEW_PATH = (
+            f"food/new?date={datetime.datetime.today().strftime('%Y-%m-%d')}&meal=0"
+        )
+        SUBMIT_POST_PATH = "food/new"
+
+        # save current date in local variable for reusing
+        date = datetime.datetime.today().strftime("%Y-%m-%d")
+
+        # get Authenticity Token
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_PATH)
+        document = self._get_document_for_url(url)
+        authenticity_token = document.xpath(
+            "(//input[@name='authenticity_token']/@value)[1]"
+        )[0]
+        utf8_field = document.xpath("(//input[@name='utf8']/@value)[1]")[0]
+
+        # submit brand and description --> Possible returns duplicates warning
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_DUPLICATE_PATH)
+        result = self.session.post(
+            url,
+            data={
+                "utf8": utf8_field,
+                "authenticity_token": authenticity_token,
+                "date": date,
+                "food[brand]": brand,
+                "food[description]": description,
+            },
+        )
+        if not result.ok:
+            raise MyfitnesspalRequestFailed(
+                f"Request Error - Unable to submit food to MyFitnessPal: status code: {result.status_code}"
+            )
+
+        # Check if a warning exists and log warning
+        document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+        if document.xpath("//*[@id='main']/p[1]/span"):
+            warning = document.xpath("//*[@id='main']/p[1]/span")[0].text
+            logger.warning(f"My Fitness Pal responded: {warning}")
+        # Passed Brand and Desc. Ready submit Form but needs new Authenticity Token
+        url = parse.urljoin(self.BASE_URL_SECURE, SUBMIT_NEW_PATH)
+        document = self._get_document_for_url(url)
+        authenticity_token = document.xpath(
+            "(//input[@name='authenticity_token']/@value)[1]"
+        )[0]
+        utf8_field = document.xpath("(//input[@name='utf8']/@value)[1]")[0]
+
+        # Step4 - Build Post Data and finally submit new Food with nutritional Details
+        data = {
+            "utf8": utf8_field,
+            "authenticity_token": authenticity_token,
+            "date": date,
+            "food[brand]": brand,
+            "food[description]": description,
+            "weight[serving_size]": serving_size,
+            "servingspercontainer": f"{servingspercontainer}",
+            "nutritional_content[calories]": f"{calories}",
+            "nutritional_content[sodium]": f"{sodium or ''}",
+            "nutritional_content[fat]": f"{fat}",
+            "nutritional_content[potassium]": f"{potassium if potassium is not None else ''}",
```

Comment:
Forgive me for the little tiny correction here -- just to explain, you're relying on the value `None` being "falsy" here, but the value `0` also is.  So if we want to be able to allow folks to specify a value of `0` for these fields (that is: not transform it into an empty string), we have to actually check if the value is `None` instead of just checking if the value is `Falsy`.  All fixed here, though, now.